internal/driver/glfw/menu_darwin.go

```diff
@@ -139,11 +139,35 @@ func exceptionCallback(e *C.char) {
 
 func handleSpecialItems(w *window, menu *fyne.Menu, nextItemID int, addSeparator bool) (*fyne.Menu, int) {
 	for i, item := range menu.Items {
-		if item.Label == "Settings" || item.Label == "Settings…" || item.Label == "Preferences" || item.Label == "Preferences…" {
+		switch item.Label {
+		case "Settings", "Settings…", "Preferences", "Preferences…":
 			items := make([]*fyne.MenuItem, 0, len(menu.Items)-1)
 			items = append(items, menu.Items[:i]...)
 			items = append(items, menu.Items[i+1:]...)
-			menu, nextItemID = handleSpecialItems(w, fyne.NewMenu(menu.Label, items...), nextItemID, false)
+			menu.Items = items
+
+			insertNativeMenuItem(C.darwinAppMenu(), item, nextItemID, 1)
+			if addSeparator {
+				C.insertDarwinMenuItem(
+					C.darwinAppMenu(),
+					C.CString(""),
+					C.CString(""),
+					C.uint(0),
+					C.int(nextItemID),
+					C.int(1),
+					C.bool(true),
+					unsafe.Pointer(nil),
+					C.uint(0),
+				)
+			}
+			nextItemID = registerCallback(w, item, nextItemID)
+		case "About":
+			items := make([]*fyne.MenuItem, 0, len(menu.Items)-1)
+			items = append(items, menu.Items[:i]...)
+			if i < len(menu.Items)-1 {
+				items = append(items, menu.Items[i+1:]...)
+			}
```

Comment:
I guess, this is a countermeasure against `i > len(menu.Items)` because the latter shrinks with the new implementation.
This reveals several flaws of the new implementation:
1. You did not apply the same protection to the old branch but only to the copied new branch.
2. You did not apply the protection to `menu.Items[i:]` which might also crash now.
3. Your `i` and `item` are no longer in sync with `menu.Items` after a change to `menu.Items` leading to unwanted effects.

Instead we should use
```go
for i := 0; i < len(menu.Items); i++ {
  item := menu.Items[i]
  …
    i-- // if special item
  …
}
```

Reply:
Thanks for catching, this should all be sorted now.